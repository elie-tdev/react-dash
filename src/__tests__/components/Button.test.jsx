import React from 'react';
import { render } from '@testing-library/react';

import {
  SuccessButton,
  PrimaryButton,
  DangerButton,
  InfoButton,
  WarningButton,
} from '@/components/Button';

test('Renders Success Button', () => {
  const { container } = render(<SuccessButton>Hello world</SuccessButton>);
  expect(container).toMatchSnapshot();
});

test('Renders Primary Button', () => {
  const { container } = render(<PrimaryButton>Hello World</PrimaryButton>);
  expect(container).toMatchSnapshot();
});

test('Renders Danger Button', () => {
  const { container } = render(<DangerButton>Hello World</DangerButton>);
  expect(container).toMatchSnapshot();
});

test('Renders Info Button', () => {
  const { container } = render(<InfoButton>Hello World</InfoButton>);
  expect(container).toMatchSnapshot();
});

test('Renders Warning Button', () => {
  const { container } = render(<WarningButton>Hello World</WarningButton>);
  expect(container).toMatchSnapshot();
});
