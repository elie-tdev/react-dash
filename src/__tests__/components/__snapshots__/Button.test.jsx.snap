// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Danger Button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root makeStyles-root-5 makeStyles-root-6 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
    data-testid="DangerButton"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Hello World
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Renders Info Button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root makeStyles-root-7 makeStyles-root-8 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
    data-testid="InfoButton"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Hello World
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Renders Primary Button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root makeStyles-root-3 makeStyles-root-4 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
    data-testid="PrimaryButton"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Hello World
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Renders Success Button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root makeStyles-root-1 makeStyles-root-2 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
    data-testid="SuccessButton"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Hello world
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Renders Warning Button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root makeStyles-root-9 makeStyles-root-10 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
    data-testid="WarningButton"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Hello World
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
